### PR TITLE
msg/async/dpdk: fix WITH_DPDK=on compilation error

### PR DESCRIPTION
**Pull request by fengchunsong**
Solve compilation errors when using with_dpdk=on

Signed-off-by: Chunsong Feng <fengchunsong@huawei.com>

### CODE REVIEW

File: src/msg/async/dpdk/DPDK.cc [hunk 1]
```diff
@@ -281,17 +276,14 @@ int DPDKDevice::init_port_start()
 
   // Set Rx VLAN stripping
   if (_dev_info.rx_offload_capa & DEV_RX_OFFLOAD_VLAN_STRIP) {
-    port_conf.rxmode.hw_vlan_strip = 1;
+     port_conf.rxmode.offloads |= DEV_RX_OFFLOAD_VLAN_STRIP;
```
Comment (rosinL) [inline src/msg/async/dpdk/DPDK.cc]: wrong ident

File: src/msg/async/dpdk/DPDK.cc [hunk 2]
```diff
@@ -188,41 +188,35 @@ int DPDKDevice::init_port_start()
...
-  }
-
-  if (!(_dev_info.tx_offload_capa & DEV_TX_OFFLOAD_TCP_TSO)) {
-    _dev_info.default_txconf.txq_flags |= ETH_TXQ_FLAGS_NOMULTSEGS;
-  }
+    // Hardware offload capabilities
+    // https://github.com/DPDK/dpdk/blob/v19.05/lib/librte_ethdev/rte_ethdev.h#L993-L1074
+    // We want to support all available offload features
+    // TODO: below features are implemented in 17.05, should support new ones
+    const uint64_t tx_offloads_wanted =
```
Comment (rosinL) [inline src/msg/async/dpdk/DPDK.cc]: wrong ident

File: src/msg/async/dpdk/DPDK.cc [hunk 3]
```diff
@@ -188,41 +188,35 @@ int DPDKDevice::init_port_start()
...
+        DEV_TX_OFFLOAD_OUTER_IPV4_CKSUM |
+        DEV_TX_OFFLOAD_QINQ_INSERT      |
+        DEV_TX_OFFLOAD_VXLAN_TNL_TSO    |
+        DEV_TX_OFFLOAD_GRE_TNL_TSO      |
+        DEV_TX_OFFLOAD_IPIP_TNL_TSO     |
+        DEV_TX_OFFLOAD_GENEVE_TNL_TSO   |
+        DEV_TX_OFFLOAD_MACSEC_INSERT;
+
+    _dev_info.default_txconf.offloads =
+        _dev_info.tx_offload_capa & tx_offloads_wanted;
```
Comment (rosinL) [inline src/msg/async/dpdk/DPDK.cc]: wrong ident

File: src/msg/async/dpdk/DPDK.cc (hunk 1 above)
Comment (rosinL) [inline src/msg/async/dpdk/DPDK.cc]: wrong ident

**Comment by fengchunsong** [comment]
use new https://github.com/rosinL/ceph/pull/18

**Pull request closed without merging**